framework/tlvf/src/src/CmduMessageTx.cpp

```diff
@@ -78,4 +78,34 @@ bool CmduMessageTx::finalize()
     if (!addClass<tlvEndOfMessage>())
         return false;
     return msg.finalize();
-};
+}
+
+/**
+ * @brief Return how much element of given size can be allocated without create a MTU restriction
+ * overflow.
+ * 
+ * @param[in] size size of one element in bytes.
+ * @return size_t number of elements of 'size' that can be inserted.
+ */
+size_t CmduMessageTx::mtu_size_restriction_overflow_check(size_t size)
+{
+    size_t current_msg_length = 0;
+    auto ptr                  = msg.prevClass();
+    auto valid_ptr            = ptr;
+    while (ptr) {
+        valid_ptr       = ptr;
+        auto class_list = ptr->getInnerClassList();
+        if (!class_list) {
+            break;
+        }
+        ptr = class_list->prevClass();
+    }
+    // auto ptr = msg.prevClass()->getInnerClassList()->prevClass();
+    if (!valid_ptr) {
```

Comment:
```suggestion
    if (!valid_ptr || !size) {
```

Reply:
Done, but added as a separate condition at the beginning of the function since there is no point doing anything if the size is 0.